src/Query.php

```diff
@@ -89,6 +81,16 @@ public function getParameter($name)
         return $this->parameters[$name];
     }
 
+    /**
+     * Returns the http client.
+     *
+     * @return \Piwik\ReportingApi\HttpClient
+     *   The http client.
+     */
+    public function getHttpClient() {
```

Comment:
The factory's responsibility is to generate an object and return it. Everything else follows that. The parameter passing and the actual execution happen in a later state.
Also, exposing this to the factory, would result in unexpected behaviour if more than one queries are generated.
I think it is fine to expose it to the client because in the end, it either should be that we expose it there, or change the architecture and have an http client that receives a query object or some query parameters and performs the request.

Reply:
:+1: